monitor-e2e/test/amb.js

```diff
@@ -0,0 +1,120 @@
+      it('should contain lastChecked', () => assert(data.lastChecked >= 0))
+      it('should contain timeDiff', () => assert(data.timeDiff >= 0))
+      it('should contain lastChecked', () => assert(data.lastChecked >= 0))
+    })
+  })
+  describe('validators', async () => {
+    let data
+
+    before(async () => {
+      ;({ data } = await axios.get(`${baseUrl}/validators`))
+    })
+
+    it('home', () => {
+      assert(typeof data.home.validators === 'object')
+      assert(data.home.validators[validator.address].balance > 0)
+      assert(data.home.validators[validator.address].leftTx > 0)
+      assert(data.home.validators[validator.address].gasPrice > 0)
+    })
+
+    it('foreign', () => {
+      assert(typeof data.foreign.validators === 'object')
+      assert(data.foreign.validators[validator.address].balance > 0)
+      assert(data.foreign.validators[validator.address].leftTx > 0)
+      assert(data.foreign.validators[validator.address].gasPrice > 0)
+    })
+
+    it('requiredSignaturesMatch', () => assert(data.requiredSignaturesMatch, 1))
+    it('validatorsMatch', () => assert(data.validatorsMatch))
+    it('lastChecked', () => assert(data.lastChecked >= 0))
+    it('timeDiff', () => assert(data.timeDiff >= 0))
+    it('homeOk', () => assert(data.homeOk))
+    it('foreignOk', () => assert(data.foreignOk))
+    it('ok', () => assert(data.ok))
+  })
+  describe('eventsStats', async () => {
+    let data
+
+    before(async () => {
+      ;({ data } = await axios.get(`${baseUrl}/eventsStats`))
+    })
+
+    it('ok', () => assert(data.ok))
+    it('lastChecked', () => assert(data.lastChecked >= 0))
+    it('timeDiff', () => assert(data.timeDiff >= 0))
+    it('home-deliveredMsgNotProcessedInForeign', () =>
+      assert(typeof data.home.deliveredMsgNotProcessedInForeign === 'object'))
+    it('home-processedMsgNotDeliveredInForeign', () =>
+      assert(typeof data.home.processedMsgNotDeliveredInForeign === 'object'))
+    it('foreign-deliveredMsgNotProcessedInHome', () =>
+      assert(typeof data.foreign.deliveredMsgNotProcessedInHome === 'object'))
+    it('foreign-processedMsgNotDeliveredInHome', () =>
+      assert(typeof data.foreign.processedMsgNotDeliveredInHome === 'object'))
+  })
+  describe('alerts', async () => {
+    let data
+
+    before(async () => {
+      ;({ data } = await axios.get(`${baseUrl}/alerts`))
+    })
+
+    it('ok', () => assert(data.ok))
+    it('lastChecked', () => assert(data.lastChecked >= 0))
+    it('timeDiff', () => assert(data.timeDiff >= 0))
+    it('executeSignatures', () => assert(typeof data.executeSignatures === 'object'))
+    it('executeAffirmations', () => assert(typeof data.executeAffirmations === 'object'))
+  })
+  describe('changing state of contracts', () => {
+    let data
+
+    before(async () => {
+      assert((await axios.get(`${baseUrl}/validators`)).data.validatorsMatch === true)
+    })
+
+    it('should change fromForeignToHomeDiff', async () => {
```

Comment:
Should we extend the test to verify catching the change in case of the Homo-to-Foreign direction?

Reply:
Yes, thanks, added the test https://github.com/poanetwork/tokenbridge/pull/231/commits/9ee0db3fba07aa57cf6fdf56834680f37fa7824d